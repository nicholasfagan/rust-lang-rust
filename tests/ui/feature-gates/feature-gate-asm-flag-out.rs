pub fn main() {
    pub fn add_catch_carry(mut a: u32, b: u32) -> u32 {
        let carry: bool;
        unsafe {
            core::arch::asm!(
                "addl {rhs:e}, {lhs:e}",
                lhs = inlateout(reg) a,
                rhs = in(reg) b,
                flag_out("c") carry,
                //~^ ERROR 9:17: 9:36: flag_out operands for inline assembly are unstable [E0658]
                options(att_syntax, nostack, nomem, pure)
            );
        }
        if carry {
            panic!("carry");
        }
        return a;
    }
    add_catch_carry(1, 2);
}
