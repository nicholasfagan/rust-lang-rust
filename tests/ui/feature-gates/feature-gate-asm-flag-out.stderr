error[E0658]: flag_out operands for inline assembly are unstable
  --> $DIR/feature-gate-asm-flag-out.rs:9:17
   |
LL |                 flag_out("c") carry,
   |                 ^^^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(asm_flag_out)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0658`.
